python_modules/dagster/dagster/_core/execution/asset_backfill.py

```diff
@@ -1247,7 +1261,24 @@ def execute_asset_backfill_iteration_inner(
 
         updated_materialized_subset = AssetGraphSubset()
         failed_and_downstream_subset = AssetGraphSubset()
+        next_latest_storage_id = instance_queryer.instance.event_log_storage.get_maximum_record_id()
     else:
+        updated_materialized_subset_data = None
```

Comment:
Move this block first to avoid a potential race condition where `parent_materialized_asset_partitions` is fetched, a materialization is written, and then `materialized_subset` is updated.

If the race condition happens, then the materialized partition's downstreams will not be requested but the partition will exist in the subset

Reply:
might be good to put this comment in code as well for discoverability in the future

Reply:
ah good call